### PR TITLE
MTDSA-195 - Get all self-employment sources (sandbox)

### PR DESCRIPTION
**Pull request by hicnar**

### CODE REVIEW

File: it/uk/gov/hmrc/selfassessmentapi/sandbox/SelfEmploymentsControllerSpec.scala [hunk 1]
```diff
@@ -34,6 +34,20 @@ class SelfEmploymentsControllerSpec extends BaseFunctionalSpec {
...
 
+    "return a valid response when retrieving list of self employments" in {
+      given()
+        .when()
+        .get(s"/sandbox/$saUtr/self-employments")
+        .thenAssertThat()
+        .statusIs(200)
+        .contentTypeIs("application/hal+json")
+        .bodyHasLink("self", s"/self-assessment/$saUtr/self-employments")
+        .bodyHasLink("self", s"/self-assessment/$saUtr/self-employments")
```
Comment (danmassie) [inline it/uk/gov/hmrc/selfassessmentapi/sandbox/SelfEmploymentsControllerSpec.scala]: same link as the line above

File: resources/public/api/documentation/1.0/Retrieve-Self-Employments.xml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+            <rows>
+                <row>
+                    <cell>
+                        <code>page</code>
+                    </cell>
+                    <cell>
+                        <code>number</code>
+                    </cell>
+                    <cell>
+                        <code>page=3</code>
```
Comment (danmassie) [inline resources/public/api/documentation/1.0/Retrieve-Self-Employments.xml]: examples should just be the value, so in this case just "3"

File: it/uk/gov/hmrc/selfassessmentapi/sandbox/SelfEmploymentsControllerSpec.scala (hunk 1 above)
Comment (hicnar) [inline it/uk/gov/hmrc/selfassessmentapi/sandbox/SelfEmploymentsControllerSpec.scala]: good eye! fixed

File: resources/public/api/documentation/1.0/Retrieve-Self-Employments.xml (hunk 2 above)
Comment (hicnar) [inline resources/public/api/documentation/1.0/Retrieve-Self-Employments.xml]: fixed

File: conf/live.routes [hunk 3]
```diff
@@ -2,6 +2,7 @@ GET        /                       uk.gov.hmrc.selfassessmentapi.controllers.liv
 GET        /:utr                   uk.gov.hmrc.selfassessmentapi.controllers.live.SelfAssessmentDiscoveryController.discover(utr: uk.gov.hmrc.domain.SaUtr)
 
 GET         /:utr/self-employments/:selfEmploymentId        uk.gov.hmrc.selfassessmentapi.controllers.live.SelfEmploymentsController.findById(utr: uk.gov.hmrc.domain.SaUtr, selfEmploymentId: uk.gov.hmrc.selfassessmentapi.domain.SelfEmploymentId)
+GET         /:utr/self-employments                          uk.gov.hmrc.selfassessmentapi.controllers.live.SelfEmploymentsController.find(utr: uk.gov.hmrc.domain.SaUtr, page: Int ?=0, pageSize: Int ?=50)
```
Comment (danmassie) [inline conf/live.routes]: Please add the link to self-employments to the links returned by the SelfAssessmentDiscoveryController

**Comment by danmassie** [comment]
:+1:

**Pull request merged**